main.go

```diff
@@ -82,6 +84,28 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
 	restConfig := ctrl.GetConfigOrDie()
+
+	var openshiftContext *utils.OpenshiftContext
+	if utils.ClusterType == utils.ClusterTypeOpenshift {
+		mcclient := mcclientset.NewForConfigOrDie(restConfig)
+		openshiftFlavor := utils.OpenshiftFlavorDefault
+		infraClient, err := client.New(restConfig, client.Options{
+			Scheme: scheme,
+		})
+		if err != nil {
+			panic(err)
+		}
+		isHypershift, err := utils.IsExternalControlPlaneCluster(infraClient)
```

Comment:
What's wrong with  `r.OpenshiftContext.IsHypershift()`

Reply:
that one only return the value it doesn't check it :p
we have it also in https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/cmd/sriov-network-config-daemon/start.go#L136

Reply:
sure I don't follow that one